insitu_plugins/src/Notification/Notification.cpp

```diff
@@ -0,0 +1,190 @@
+#include <Notification/Notification.hpp>
+#include <Notification/Notification_dialog.hpp>
+#include <insitu_utils/painter.hpp>
+
+using namespace notification_filter;
+
+namespace insitu_plugins
+{
+/*
+    Filter Implementation
+*/
+
+Notification::Notification(void)
+{
+    // TODO instantiation code
+}
+
+void Notification::filterInit(void)
+{
+    settingsDialog = new NotificationDialog(this);
+    setSize(QSize(300, 300));
+
+    nh_ = getNodeHandle();
+
+    topic_name_ = getSettingsValue().get("topic", DEFAULT_TOPIC).asString();
+
+    topic_subscriber_ =     // 2nd parameter is queue size
+        nh_.subscribe(topic_name_, 1, &Notification::handleCallback, this);
+    
+    queue_size_ = getSettingsValue().get("queue_size", std::stoi(DEFAULT_QUEUE_SIZE)).asInt();
```

Comment:
Need this to build
```suggestion
    queue_size_ = getSettingsValue().get("queue_size", DEFAULT_QUEUE_SIZE).asInt();
```

Reply:
Must be within /insitu_utils/src/painter.cpp.

I had to play around with the marginx, marginy, and scale variables a bit in order to make the text work whenever the "number of messages" displayed changed. Now, since the topic_name is also printed in the text box, it is definitely changing those parameters which is why you're seeing the text shift in y-direction.

Guess I'll have to spend some more time coming up with a more robust calculation that considers all the possible variables. Or maybe the better option might be to default to a value that shows appropriately and add a zoom functionality.

Either way, I'll need some time because this one looks to be more challenging than the previous changes.

Reply:
I believe `DEFAULT_QUEUE_SIZE` is an int to begin with, so need for `std:stoi` just as Adam mentions.

On a more function-oriented note, you as the filter implementer have agency over the size of the filter as well. You can actually just make the filter area taller as more messages show up by modifying the QSize child. Alternatively, if going with a fixed size, you can simply make it scroll and dismiss the oldest notification as new ones come in (`std::deque` is useful here).